app/objects/c_operation.py

```diff
@@ -115,6 +116,7 @@ def set_start_details(self):
 
     def add_link(self, link):
         self.chain.append(link)
+        Event('link added')
```

Comment:
I think we should make this subject, verb/action so that a service can subscribe to all 'link' actions

Reply:
this will also reduce look up times